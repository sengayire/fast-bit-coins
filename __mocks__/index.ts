jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {})
      }
    }
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {}
  }
}))

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn()
}))
