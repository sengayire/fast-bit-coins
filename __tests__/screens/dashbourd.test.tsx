import { NavigationContainer } from '@react-navigation/native'
import { render } from '@testing-library/react-native'

import '../../__mocks__'
import '@testing-library/jest-native/extend-expect'
import Dashboard from 'src/screens/dashboard'

describe('Dashboard component', () => {
  it('Should render dashboard', () => {
    const view = render(
      <NavigationContainer>
        <Dashboard />
      </NavigationContainer>
    )
    expect(view).toMatchSnapshot()
  })
})
