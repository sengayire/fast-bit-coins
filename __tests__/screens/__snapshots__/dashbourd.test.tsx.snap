// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Should render dashboard 1`] = `
<View>
  <ViewManagerAdapter_ExpoImage
    borderBottomLeftRadius={200}
    height="80%"
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    proxiedProperties={
      {
        "contentFit": "cover",
        "contentPosition": {
          "left": "50%",
          "top": "50%",
        },
        "placeholder": [],
        "source": [],
        "tintColor": undefined,
        "transition": null,
      }
    }
    style={
      {
        "borderBottomLeftRadius": 200,
        "height": "80%",
        "width": "100%",
      }
    }
    width="100%"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "position": "absolute",
        "right": 20,
        "top": 80,
      }
    }
  >
    <ViewManagerAdapter_ExpoImage
      height={30}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      proxiedProperties={
        {
          "contentFit": "cover",
          "contentPosition": {
            "left": "50%",
            "top": "50%",
          },
          "placeholder": [],
          "source": [],
          "tintColor": undefined,
          "transition": null,
        }
      }
      style={
        {
          "height": 30,
          "width": 30,
        }
      }
      width={30}
    />
    <Text>
      dashboard.logout
    </Text>
  </View>
  <View
    style={
      {
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 30,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      dashboard.title
    </Text>
    <View>
      <Text
        style={
          [
            {
              "color": "#707070",
            },
            {
              "fontSize": 14,
              "fontWeight": 400,
              "lineHeight": 18,
              "paddingHorizontal": 48,
              "paddingVertical": 19,
              "textAlign": "center",
            },
          ]
        }
      >
        dashboard.text
      </Text>
    </View>
    <View
      style={
        {
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          {
            "color": "#FFB81C",
            "fontSize": 15,
            "textAlign": "right",
          }
        }
      >
        dashboard.more
      </Text>
    </View>
  </View>
</View>
`;
